Source/WebCore/platform/graphics/cg/ImageUtilitiesCG.cpp

```diff
@@ -125,4 +125,24 @@ Vector<String> transcodeImages(const Vector<String>& paths, const String& destin
     });
 }
 
+std::pair<String, Vector<IntSize>> utiAndAvailableSizesFromImageData(std::span<const uint8_t> data)
+{
+    Ref buffer = FragmentedSharedBuffer::create(data);
+    Ref imageDecoder = ImageDecoderCG::create(buffer.get(), AlphaOption::Premultiplied, GammaAndColorProfileOption::Applied);
+    imageDecoder->setData(buffer.get(), true);
+    auto uti = imageDecoder->uti();
+    if (!isSupportedImageType(imageDecoder->uti()))
+        return { { }, { } };
+
+    // No support for animated image.
+    if (imageDecoder->repetitionCount() == RepetitionCountInfinite)
+        return { uti, { } };
```

Comment:
Comment doesn’t match the code; images with repetition counts like 2 are also animated images.

Reply:
I think repetition count 2 is infinite since we only have None, Once and Infinite:
https://github.com/WebKit/WebKit/blob/f607bbdba4df616b144d16367b4f4fa3ee329a02/Source/WebCore/platform/graphics/ImageTypes.h#L55

Reply:
I believe `Once` is also an animated image.

Reply:
I found gif could have 1 frame with loop count 1 and I think we need to compute size for that case. To make it clear, I updated this check to return early if loop count is not 0 and it contains multiple frames.